api/src/main/java/org/pentaho/platform/api/genericfile/GetTreeOptions.java

```diff
@@ -148,6 +184,36 @@ public void setExpandedPath( @Nullable GenericFilePath expandedPath ) {
     this.expandedPath = expandedPath;
   }
 
+  /**
+   * Sets the tree filter.
+   * If a null value is passed, "ALL" filter will be used.
+   * @param filter
+   */
+  public void setFilter( TreeFilter filter ) {
+    if ( filter == null ) {
+      this.filter = TreeFilter.ALL;
+    } else {
+      this.filter = filter;
+    }
+  }
+
+  /**
+   * Sets the tree filter via parsing string value.
+   * @param treeFilterString
+   * @throws IllegalArgumentException if an invalid value is passed.
+   */
+  public void setFilter( String treeFilterString ) throws IllegalArgumentException {
+    setFilter( GetTreeOptions.TreeFilter.valueOf( treeFilterString ) );
```

Comment:
Should also support the `null` case in this overload.
Also, the `throws IllegalArgumentException` is not needed as it is not checked. I know I misled you to include it...

Reply:
Why shouldn't it `throws IllegalArgumentException` it is possible for someone to make a request with a bad filter that will result in this exception from `valueOf`. Our code shouldn't do that, but if we somehow do by accident, I think it is good to throw and account for this exception.

Reply:
What I meant is that, afaik, it is not needed to add a `throws IllegalArgumentException` clause to the method, because it is not a checked exception. You would still throw the exception.

Reply:
Talked offline and it's ok to declare locally thrown runtime exceptions, if they're not obvious (e.g. NPE) or otherwise provide value to the documentation.